app/controllers/tag_implication_controller.rb

```diff
@@ -1,76 +1,96 @@
+# frozen_string_literal: true
+
 class TagImplicationController < ApplicationController
-  layout "default"
-  before_action :member_only, :only => [:create]
+  layout 'default'
+  before_action :member_only, only: [:create]
 
   def create
-    ti = TagImplication.new(tag_implication_params.merge(:is_pending => true, :creator_id => @current_user.id))
-
-    if ti.save
-      flash[:notice] = "Tag implication created"
-    else
-      flash[:notice] = "Error: " + ti.errors.full_messages.join(", ")
-    end
-
-    redirect_to :action => "index"
+    ti = TagImplication.new(tag_implication_params.merge(is_pending: true, creator_id: @current_user.id))
+    flash[:notice] = ti.save ? 'Tag implication created' : "Error: #{ti.errors.full_messages.join(', ')}"
+    redirect_to action: 'index'
   end
 
   def update
     ids = params[:implications].try(:keys)
 
     case params[:commit]
-    when "Delete"
-      if @current_user.is_mod_or_higher? || TagImplication.where(:id => ids, :is_pending => true, :creator_id => @current_user.id).count == ids.count
-        ids.each { |x| TagImplication.find(x).destroy_and_notify(@current_user, params[:reason]) }
-
-        flash[:notice] = "Tag implications deleted"
-        redirect_to :action => "index"
-      else
-        access_denied
-      end
-
-    when "Approve"
-      if @current_user.is_mod_or_higher?
-        ids.each do |x|
-          if CONFIG["enable_asynchronous_tasks"]
-            JobTask.create(:task_type => "approve_tag_implication", :status => "pending", :data => { "id" => x, "updater_id" => @current_user.id, "updater_ip_addr" => request.remote_ip })
-          else
-            TagImplication.find(x).approve(@current_user.id, request.remote_ip)
-          end
-        end
-
-        flash[:notice] = "Tag implication approval jobs created"
-        redirect_to :controller => "job_task", :action => "index"
-      else
-        access_denied
-      end
+    when 'Delete'
+      delete_tags(ids)
+    when 'Approve'
+      approve_tags(ids)
     else
       head :bad_request
     end
   end
 
   def index
-    if params[:commit] == "Search Aliases"
-      return redirect_to :controller => "tag_alias", :action => "index", :query => params[:query]
-    end
+    return redirect_to_tag_alias_index if searching_aliases?
+
+    @implications = TagImplication.retrieve_matching_tag_id
+    @implications = TagImplication.apply_query_filter(params, @implications)
+    @implications = @implications.paginate(page: page_number, per_page: 20)
```

Comment:
proper [scope](https://guides.rubyonrails.org/active_record_querying.html#scopes) can be chained:

```suggestion
    @implications = TagImplication
      .order_for_listing # it's only sorting, there's no "retrieve_matching_tag_id" in it
      .search_by_tag_name(params[:query]) # don't pass the whole params
      .paginate(page: page_number, per_page: 20)
```

Reply:
Thank you for the suggestion! I have applied this and made adjustments to `tag_implication.rb`